File: app/src/main/java/com/example/testgithub/db/OrgRepDao.java
package com.example.testgithub.db;


import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import com.example.testgithub.model.OrgRepos;

import java.util.List;

@Dao
public abstract class OrgRepDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    public abstract void insertOrgRepos(List<OrgRepos> orgRepos);

    @Query("SELECT * FROM OrgRepos")
    public abstract LiveData<List<OrgRepos>> loadOrgRepositories();

}

File: app/src/main/java/com/example/testgithub/db/OrgRepDao.kt
package com.example.testgithub.db


import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.example.testgithub.model.OrgRepos

@Dao
abstract class OrgRepDao {


    @Insert(onConflict = OnConflictStrategy.REPLACE)
    abstract fun insertOrgRepos(orgRepos: List<OrgRepos>)


    @Query("SELECT * FROM OrgRepos")
    abstract fun loadOrgRepositories(): LiveData<List<OrgRepos>>

}
